auth/server/src/main/java/com/seagates3/controller/AccountController.java

```diff
@@ -171,6 +190,22 @@ public ServerResponse create() {
           LOGGER.error("Create account delay failing - ", e);
           Thread.currentThread().interrupt();
         }
+        // Handle multi-threaded create API calls by deleting extra accounts,
+        // if account limit creation exceeded due to multiple thread/multiple
+        // node create() API calls.
+        try {
+          accountCount = accountDao.getTotalCountOfAccounts();
+
+          if (accountCount > maxAllowedLdapResults) {
+            // delete newly created account since we exceeded account
+            // creation limit.
+            accountDao.ldap_delete_account(account);
+            return accountResponseGenerator.internalServerError();
```

Comment:
This should return error message saying "Account limit exceeded"

Reply:
lets say we have 999 account and two threads parallelly try to do createAccount() .
1)  Since both thread starts same time they will add entries in ldap e.g. thread-1 adds 1000th account and thread-2 adds 1001th account.
2) at this point threads will check total ldap entry count and both receives value greater than 1000 hence they will delete their newly created accounts i.e. 1000th and 1001th . 
3) in case of successful deletion we will still have 999 entries in ldap ..
Hence we  should not send "Account limit exceed" message since both newly created account got deleted.
so that next time client can send another request which will create 1000th account.

Reply:
what's the error message sent to client in this case

Reply:
it will be InternalServer Error